src/storage.rs

```diff
@@ -0,0 +1,43 @@
+use std::sync::Arc;
+
+const MAX_INLINE: usize = 39;
```

Comment:
why 39?

Reply:
So the total size is 40, a multiple of 8. 1 byte is needed for the enum discriminator.

Reply:
39 fits all 256 bit hashes with some room to spare. 512 byte hashes won't work though, and I think once you go to such large hashes, you are better off with an `Arc<[u8]>`.

But the size of the inline buffer can of course be adjusted when even larger hashes become common. The whole mechanism is completely opaque.